### PR TITLE
Module Configuration Import/Export

### PR DESCRIPTION
**Pull request by pkendall64**
# Problem
I've got my models all setup in my ELRS module (all 64 of them 😉), and I just got a new module/radio and I don't want to go through the hassle of setting them all up again!
Help me out devs!

# Solution
We got you back bro!
On the "Options" tab we've added a new section, "Import/Export", which allow you to save your module config, along with all your models to a JSON file. This file can then be loaded into another module.

Screenshot 2022-11-13 at 4 08 40 PM

The information saved in the file is as follows:
- Fan mode
- Power fan threshold
- Motion mode
- VTX admin
   - Band
   - Channel
   - Pitmode
   - Power
- Backpack
   - DVR start delay
   - DVR stop delay
   - DVR aux channel

All 64 models:
- Packet rate
- Telemetry ratio
- Switch mode
- Power settings
   - Max power
   - Dynamic power
   - Boost channel
- Model match enabled flag
- TX antenna

# Extra
Converted the link to reset options to a button as well, so it's consistent.

### CODE REVIEW

**Comment by JyeSmith** [comment]
Could this be extended to be a generic clone function e,g, also include bind phrase, wifi details etc?

**Comment by pkendall64** [comment]
> Could this be extended to be a generic clone function e,g, also include bind phrase, wifi details etc?

Yes, we could add that too. We'd probably then want the user to select which items to export and/or import i.e. models and options or just one.

File: src/html/scan.js [hunk 1]
```diff
@@ -349,6 +349,36 @@ _('upload_form').addEventListener('submit', (e) => {
...
+          cuteAlert({
+            type: 'info',
+            title: 'Upload Model Configuration',
+            message: this.responseText
+          });
+        } else {
+          cuteAlert({
+            type: 'error',
+            title: 'Upload Model Configuration',
+            message: 'An error occurred while upload model configuration file'
```
Comment (deadbytefpv) [inline src/html/scan.js:370]: An error occurred while uploading model configuration file.

**Review by deadbytefpv — CHANGES_REQUESTED** [review]
Tested importing config from a NamimnoRC Flash OLED TX into an EMAX 2.4G OLED TX, works great.

Tested importing config from a NamimnoRC Flash OLED TX into an EMAX 900 OLED TX, the import works, but because it's a 2.4GHz TX config being saved into a 900MHz TX, the parameters will not always line up. See attached screengrab.
IMG_0304

**Review by JyeSmith — COMMENTED** [review]
Tested with the Emax 900 OLED.  Changed all the things and exported the settings.  Then changed everything and imported the setting which successfully restored the original settings.

**Review by JyeSmith — APPROVED** [review]

**Review by deadbytefpv — APPROVED** [review]

**Pull request merged**